pkg/roachpb/errors.go

```diff
@@ -49,6 +49,15 @@ func (e *RetryableTxnError) Error() string {
 
 var _ error = &RetryableTxnError{}
 
+// NewLocalRetryableError creates a shim RetryableTxnError that
+// reports the given cause when converted to String(). This can be
+// used to fake/force a retry at the SQL layer.
+func NewLocalRetryableError(cause string) *RetryableTxnError {
```

Comment:
I don't think in Go you can initialize unexported members of a struct in a different package, i.e. `&roachpb.RetryableTxnError{message: cause}` in package `sql/parser` will complain that `message` is unexported and thus cannot be used directly.